cache.go

```diff
@@ -272,11 +273,27 @@ func (c *BuildCache) Lookup(name string) (*CacheEntry, bool, error) {
 			return nil, false, err
 		}
 		if dirChanged {
-			log.Infof("cache miss because overlay_dir content changed: %s", overlayDir.Source)
-			return nil, false, nil
+			overlayDirsChanged = true
+			changedOverlayDirs = append(changedOverlayDirs, overlayDir.Source)
 		}
 	}
-
+	if overlayDirsChanged {
+		log.Infof("cache miss because content of %d overlay dirs changed:", len(changedOverlayDirs))
+		if c.config.Debug {
+			for _, dir := range changedOverlayDirs {
+				log.Infof(dir)
+			}
+		} else {
+			for index, dir := range changedOverlayDirs {
+				log.Infof(dir)
+				if index == 1 && len(changedOverlayDirs) > 2 {
```

Comment:
Move the "if c.config.Debug" check here, since you are doing log.Infof(dir) in both cases

Reply:
Modified as requested